cornellh4i/lagos-volunteers
# Event registration fullstack

## Summary


1. Modified `/pages/events/[eventid]/register`, `/pages/events/[eventid]/cancel`, `/pages/events/[eventid]/confirm/register`, `/pages/events/[eventid]/confirm/cancel` to use data from a GET request instead of dummy data.
2. Integrated backend functionality with Accept & Register in the event modal. User is now registered for 
the event and is redirected to the confirmation page.
3. Integrated backend functionality with Cancel Registration in the event modal. User is now canceled for 
 the event and is redirected to the confirmation page.
4. Implemented email notification after registering/cancelling an event trough SendGrid API.

Closes #94

## Testing


Tools needed: 
Prisma DB

1. Pull from branch `event-registration-fullstack`
2. Executie in your terminal `yarn add @sendgrid/mail`
3. Run `yarn run build` and `yarn run start`
4. Open up `localhost:5555`. This should be where the Prisma DB is.
5. Open up `localhost:3000`. This should be where the website. Sign in as 'prisma@hey.com' or any other user you'd like.
6. Open up the "User" model and search for the user 'prisma@hey.com' (If on the website you signed in as a different user look for that user instead). Note the ID for that user!

<img width="669" alt="image" src="https://github.com/cornellh4i/lagos-volunteers/assets/90882246/22b23123-8a5b-4b68-b798-c28deaee3453">

8. Open up the "Event" model and get any Event ID in Prisma DB.
<img width="669" alt="image" src="https://github.com/cornellh4i/lagos-volunteers/assets/90882246/24120bfd-2877-4e2b-bce6-0671faef6e69">

9. Paste the Event ID you got from step 8. Now look up the URL `http://localhost:3000/events/<eventid>/register` and register the user. 
<img width="951" alt="image" src="https://github.com/cornellh4i/lagos-volunteers/assets/90882246/8f113c0c-47f5-41a4-a435-d971d63baa11">

10. To verify that you've registered a user go into the "EventEnrollment" model in Prisma and search for the eventID and ensure that the userID that you stored earlier is assigned to that event.
<img width="1084" alt="image" src="https://github.com/cornellh4i/lagos-volunteers/assets/90882246/c18ef90d-d9f7-4c48-a03a-00678a5f915e">

11. Now look up the URL `http://localhost:3000/events/<eventid>/cancel` and cancel the user.
<img width="914" alt="image" src="https://github.com/cornellh4i/lagos-volunteers/assets/90882246/c7137cd5-281d-4a9f-93fc-052b3b085303">

12. To verify that you've canceled a user go into the "EventEnrollment" model in Prisma and you should no longer see this event assigned to that user. 

## Notes
Dan: I left in some console.log statements so the tester can follow the flow of the requests in case an error occurs. They'll be removed in final review.

Louis: I found this frontend issue as shown below.

<img width="779" alt="Screenshot 2023-10-22 at 20 54 09" src="https://github.com/cornellh4i/lagos-volunteers/assets/94189315/3147c2ae-98f8-4ee0-a4d6-c225fb25f76c">
